### PR TITLE
Add support for entities with role and group labels

### PR DESCRIPTION
**Pull request by tabergma**
**Proposed changes**:
- Updated `DIETClassifier` and `CRFEntityExtractor` to support entities with role and group labels
- Unified way how entities are handled in `DIETClassifier` and `CRFEntityExtractor` 
- Update training data format to support entities with roles/groups

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: rasa/nlu/classifiers/diet_classifier.py [hunk 1]
```diff
@@ -1426,34 +1510,93 @@ def batch_loss(
...
+    def _batch_loss_entities(
+        self,
+        mask_text: tf.Tensor,
+        sequence_lengths: tf.Tensor,
+        text_transformed: tf.Tensor,
+        tf_batch_data: Dict[Text, List[tf.Tensor]],
+    ) -> List[tf.Tensor]:
+        losses = []
+
+        sequence_lengths = sequence_lengths - 1  # remove cls token
```
Comment (federicotdn) [inline rasa/nlu/classifiers/diet_classifier.py:1558]:
```suggestion
        sequence_lengths -= 1 # remove cls token
```
😄

File: rasa/nlu/extractors/extractor.py [hunk 2]
```diff
@@ -333,3 +368,182 @@ def filter_trainable_entities(
...
+            tag_names: The tag names to include in the entity.
+            entity_tag: The entity type value.
+            group_tag: The entity group value.
+            role_tag: The entity role value.
+            token: The token.
+            confidence: The confidence value.
+
+        Returns:
+            Created entity.
+
```
Comment (federicotdn) [inline rasa/nlu/extractors/extractor.py:498]:
```suggestion
```

File: rasa/nlu/extractors/extractor.py [hunk 3]
```diff
@@ -333,3 +368,182 @@ def filter_trainable_entities(
...
+            entity[ENTITY_ATTRIBUTE_GROUP] = group_tag
+            if confidences is not None:
+                entity[ENTITY_ATTRIBUTE_CONFIDENCE_GROUP] = confidences[
+                    ENTITY_ATTRIBUTE_GROUP
+                ][idx]
+
+        return entity
+
+    @staticmethod
+    def check_correct_entity_annotations(training_data: TrainingData) -> None:
```
Comment (federicotdn) [inline rasa/nlu/extractors/extractor.py:527]: We should add a docstring for this public method.

File: rasa/nlu/test.py [hunk 4]
```diff
@@ -833,30 +836,48 @@ def find_intersecting_entites(token: Token, entities: List[Dict]) -> List[Dict]:
...
+    entities.
+
+    Args:
+        token: a single token
+        candidates: entities found by a single extractor
+        attribute_key: the attribute key of interest
 
+    Returns:
+        the value of the attribute key of the best fitting entity
+
```
Comment (federicotdn) [inline rasa/nlu/test.py:855]:
```suggestion
```

File: rasa/nlu/test.py [hunk 5]
```diff
@@ -912,6 +933,42 @@ def align_entity_predictions(
     }
 
 
+def _concat_entity_labels(
+    token: Token, entities: List[Dict], extractors: Optional[Set[Text]] = None
+):
```
Comment (federicotdn) [inline rasa/nlu/test.py:938]:
```suggestion
) -> Text:
```

File: rasa/nlu/test.py [hunk 6]
```diff
@@ -1082,8 +1140,18 @@ def get_available_response_selector_types(interpreter: Interpreter) -> List[Text
...
+
+    Remove pre-trained extractors so that entities from pre-trained extractors
+    are not predicted upon parsing.
+
+    Args:
+        pipeline: the pipeline
+
+    Returns:
+        Updated pipeline
+
```
Comment (federicotdn) [inline rasa/nlu/test.py:1153]:
```suggestion
```

File: rasa/nlu/utils/__init__.py [hunk 7]
```diff
@@ -56,13 +64,43 @@ def write_to_file(filename: Text, text: Any) -> None:
...
+        end: end position of entity
+        value: text value of the entity
+        entity_type: name of the entity type
+        role: role of the entity
+        group: group of the entity
+        **kwargs: additional parameters
+
+    Returns:
+        an entity dictionary
+
```
Comment (federicotdn) [inline rasa/nlu/utils/__init__.py:90]:
```suggestion
```

File: rasa/version.py [hunk 8]
```diff
@@ -1,3 +1,3 @@
 # this file will automatically be changed,
 # do not add anything but the version number here!
-__version__ = "1.9.6"
+__version__ = "1.9.0"
```
Comment (federicotdn) [inline rasa/version.py:3]: Was this change intentional?

File: data/test/demo-rasa-composite-entities.md [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+**Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*
+
+- [intent:affirm](#intentaffirm)
+- [intent:goodbye](#intentgoodbye)
+- [intent:greet](#intentgreet)
+- [intent:chitchat](#intentchitchat)
+- [intent:book_flight](#intentbook_flight)
+- [intent:order_pizza](#intentorder_pizza)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
```
[the remark this answers is not in the document]
Comment (tabergma) [reply data/test/demo-rasa-composite-entities.md:12]: I tried to remove it, but it keeps getting added again. How can I remove it so that it does not appear again?

[the remark this answers is not in the document]
Comment (federicotdn) [reply data/test/demo-rasa-composite-entities.md:12]: You can try temporarily removing the `.git/hooks/pre-commit` file before commiting

File: rasa/version.py (hunk 8 above)
Comment (tabergma) [reply rasa/version.py:3]: no, will change it back
